packages/flutter_tools/test/general.shard/generate_localizations_test.dart

```diff
@@ -2647,6 +2647,36 @@ import 'output-localization-file_en.dart' deferred as output-localization-file_e
           )),
         );
       });
+
+      testWithoutContext('cannot have empty key', () {
+        const String emptyKeyArbFile = '''
+{
+  "": "value of an empty key"
+}''';
+        final Directory l10nDirectory = fs.currentDirectory.childDirectory('lib').childDirectory('l10n')
+          ..createSync(recursive: true);
+        l10nDirectory.childFile(defaultTemplateArbFileName)
+            .writeAsStringSync(emptyKeyArbFile);
+
+        expect(
+              () {
+            LocalizationsGenerator(
```

Comment:
nit: didn't see this before, but this one should also be unindented:

```suggestion
        expect(
          () {
            LocalizationsGenerator(
```